apps/system/test/marionette/task_manager_test.js

```diff
@@ -125,6 +128,55 @@ marionette('Task Manager', function() {
     });
   });
 
+  suite('when launched while an app is still initializing', function() {
+    var slowApp;
+    setup(function() {
+      // launch a 3rd app which we can mock easily
+      slowApp = new FakeApp(client, 'app://' + slowAppOrigin);
+      slowApp.launch();
+
+      var iframeId = slowApp.iframe.getAttribute('id');
+      // mock iframe.getScreenshot so the screenshot never appears
```

Comment:
not a big fan of mocking in integration tests but I don't see how to test this without triggering nightmarish intermittents without it :)